graspologic/align/seedless_procrustes.py

```diff
@@ -319,12 +319,11 @@ def _compute_objective(
         Y: np.ndarray,
         Q: Optional[np.ndarray] = None,
         P: Optional[np.ndarray] = None,
-    ) -> Union[float, np.ndarray]:
-        if Q is None:
-            Q = self.Q_
-        if P is None:
-            P = self.P_
-        return np.linalg.norm(X @ Q - P @ Y, ord="fro")
+    ) -> np.floating:
```

Comment:
1: numpy added a 'floating' type that means 'float or a numpy array of floats', which is really what we meant anyway
2: we had some bad habits in the code where we define a variable as bound to a specific type (namely, Optional), and then reset it to something that is decidedly not optional.  I follow the _VariableName pattern pretty much throughout this PR